src/postings/segment_postings.rs

```diff
@@ -98,6 +103,81 @@ impl<'a> DocSet for SegmentPostings<'a> {
         }
     }
 
+    fn skip_next(&mut self, target: DocId) -> SkipResult {
+        if !self.advance() {
+            return SkipResult::End;
+        }
+
+        let mut pos = self.index_within_block();
+        // skip blocks until one that might contain the target
+        loop {
+            // check if we need to go to the next block
+            if target > self.block_decoder.output(self.block_len - 1) {
+                self.cur += Wrapping(self.block_len - pos);
+                self.load_next_block();
+                pos = 0;
+
+                // there was no more data
+                if self.block_len == 0 {
+                    return SkipResult::End;
```

Comment:
I need to double check that BlockDecoder really has that kind of contracts.

Alternatively a comparison of `self.len` and `self.cur.0`